### ISSUE TITLE
This cycle occurred because of a configuration option

### ISSUE THREAD

**Issue by pseudotensor** [NVIDIA/tensorflow#21]
docker build of latest r1.15.5+nv21.03 leads to:

```
WARNING: /tfbuild/tensorflow/tensorflow/core/BUILD:1098:12: in srcs attribute of cc_library rule //tensorflow/core:framework_lite: please do not import '//tensorflow/core/platform:default/integral_types.h' directly. You should either move the file to this package or depend on an appropriate rule there
WARNING: /tfbuild/tensorflow/tensorflow/core/BUILD:1098:12: in srcs attribute of cc_library rule //tensorflow/core:framework_lite: please do not import '//tensorflow/core/platform:default/mutex.h' directly. You should either move the file to this package or depend on an appropriate rule there
WARNING: /tfbuild/tensorflow/tensorflow/core/BUILD:1098:12: in srcs attribute of cc_library rule //tensorflow/core:framework_lite: please do not import '//tensorflow/core/platform:default/mutex_data.h' directly. You should either move the file to this package or depend on an appropriate rule there
WARNING: /tfbuild/tensorflow/tensorflow/core/BUILD:1096:1: in linkstatic attribute of cc_library rule //tensorflow/core:framework_lite: setting 'linkstatic=1' is recommended if there are no object files
WARNING: /tfbuild/tensorflow/tensorflow/core/BUILD:388:12: in srcs attribute of cc_library rule //tensorflow/core:platform_port: please do not import '//tensorflow/core/platform:cpu_info.cc' directly. You should either move the file to this package or depend on an appropriate rule there
WARNING: /tfbuild/tensorflow/tensorflow/core/BUILD:388:12: in srcs attribute of cc_library rule //tensorflow/core:platform_port: please do not import '//tensorflow/core/platform:default/dynamic_annotations.h' directly. You should either move the file to this package or depend on an appropriate rule there
WARNING: /tfbuild/tensorflow/tensorflow/core/BUILD:388:12: in srcs attribute of cc_library rule //tensorflow/core:platform_port: please do not import '//tensorflow/core/platform:default/mutex.h' directly. You should either move the file to this package or depend on an appropriate rule there
WARNING: /tfbuild/tensorflow/tensorflow/core/BUILD:388:12: in srcs attribute of cc_library rule //tensorflow/core:platform_port: please do not import '//tensorflow/core/platform:posix/port.cc' directly. You should either move the file to this package or depend on an appropriate rule there
WARNING: /tfbuild/tensorflow/tensorflow/core/BUILD:652:12: in srcs attribute of cc_library rule //tensorflow/core:lib_proto_parsing: please do not import '//tensorflow/core/platform:protobuf.cc' directly. You should either move the file to this package or depend on an appropriate rule there
WARNING: /tfbuild/tensorflow/tensorflow/core/BUILD:2460:12: in srcs attribute of cc_library rule //tensorflow/core:lib_internal_impl: please do not import '//tensorflow/core/platform:abi.h' directly. You should either move the file to this package or depend on an appropriate rule there
WARNING: /tfbuild/tensorflow/tensorflow/core/BUILD:2460:12: in srcs attribute of cc_library rule //tensorflow/core:lib_internal_impl: please do not import '//tensorflow/core/platform:blocking_counter.h' directly. You should either move the file to this package or depend on an appropriate rule there
```

and

```
ERROR: /tfbuild/tensorflow/tensorflow/core/BUILD:2827:1: in cc_library rule //tensorflow/core:framework_internal: cycle in dependency graph:
    //tensorflow/tools/pip_package:build_pip_package
    //tensorflow/python/compiler:compiler
    //tensorflow/python/compiler/tensorrt:init_py
    //tensorflow/python/compiler/tensorrt:trt_convert_py
    //tensorflow/python/saved_model:save
    //tensorflow/python/training/tracking:graph_view
    //tensorflow/python:framework_ops
    //tensorflow/python:type_spec
    //tensorflow/python:tensor_shape
    //tensorflow/python/eager:monitoring
    //tensorflow/python:c_api_util
    //tensorflow/python:pywrap_tensorflow
    //tensorflow/python:pywrap_tensorflow_internal
    //tensorflow/python:pywrap_tensorflow_internal.py
    //tensorflow/python:pywrap_tensorflow_internal_py_wrap
    //tensorflow/core/distributed_runtime:server_lib
.-> //tensorflow/core:framework_internal
|   //tensorflow/core:framework_internal_impl
|   //tensorflow/core/profiler:nvtx_utils
|   //tensorflow/core:framework
`-- //tensorflow/core:framework_internal
This cycle occurred because of a configuration option
ERROR: Analysis of target '//tensorflow/tools/pip_package:build_pip_package' failed; build aborted
INFO: Elapsed time: 42.064s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (398 packages loaded, 20384 targets configured)
make: *** [build_tensorflow_one] Error 1
[root@3d36a0da2e0c tfbuild]# 
```

Same thing does not happen with 1.15.4+nv20.12

**Comment by pseudotensor** [comment]
Building like this in nvidia docker, to get CPU version of build:
```
    TEST_TMPDIR=/tmp/ PYTHON_BIN_PATH=/root/.pyenv/versions/3.8.5/bin/python \
    PYTHON_LIB_PATH=/root/.pyenv/versions/3.8.5/lib/python3.8/site-packages \
    GCC_HOST_COMPILER_PATH=/opt/rh/devtoolset-7/root/usr/bin/gcc TF_NEED_MPI=0 TF_SET_ANDROID_WORKSPACE=0 \
    TF_NEED_GCP=1 TF_NEED_HDFS=1 TF_NEED_OPENCL=0 TF_NEED_CUDA=0 \
    TF_ENABLE_XLA=1 TF_NEED_TENSORRT=0 TF_NEED_ROCM=0 TF_NEED_OPENCL_SYCL=0 TF_CUDA_CLANG=0 CC_OPT_FLAGS="-march=native" \
    TF_DOWNLOAD_CLANG=0 \
    TF_CUDA_VERSION=11 TF_CUDNN_VERSION=8 TF_CUDA_COMPUTE_CAPABILITIES="3.5, 6.0, 7.0" python ./configure.py && \
    bazel build --config=v1 --config=monolithic //tensorflow/tools/pip_package:build_pip_package && \
    ./bazel-bin/tensorflow/tools/pip_package/build_pip_package /tfbuild/

```

**Comment by nluehr** [comment]
The build can succeed if the option `--config=monolithic` is omitted.

**Comment by nluehr** [comment]
This is fixed in the [22.01 release](https://github.com/NVIDIA/tensorflow/releases/tag/v1.15.5%2Bnv22.01).
